データベース設計の基本

john-2132 opened the pull request:
## 課題のリンク

* https://github.com/happiness-chain/practice/blob/main/database/design.md

## やったこと

* twitter.erの作成と、erdの生成、コミット。


## 確認方法

* 生成物であるpdfファイルを確認し、テーブルの情報、関連が正しいこと。

john-2132 committed:
データベース設計の基本
twitter.erdの作成と、その生成物をコミット

yamaryo416 commented on twitter_erd/twitter.er:
```diff
@@ -0,0 +1,71 @@
+[users]
+  *user_id {label: "bigint, not null"}
```
各テーブルも同様に`id`としましょう。

yamaryo416 commented on twitter_erd/twitter.er:
```diff
@@ -0,0 +1,71 @@
+[users]
+  *user_id {label: "bigint, not null"}
+  user_name {label: "varchar, not null, uniq"}
+  e_mail {label: "varchar, not null, uniq"}
```
こちら`email`の方が良いですね。

yamaryo416 commented on twitter_erd/twitter.er:
```diff
@@ -0,0 +1,71 @@
+
+[direct_messages] {bgcolor: "#e6e6fa"}
+  *dm_id {label: "bigint, not null"}
+  dm_text {label: "varchar, not null"}
```
こちら`text`の方が良いですね。

yamaryo416 commented on twitter_erd/twitter.er:
```diff
@@ -0,0 +1,71 @@
+
+[tweets] {bgcolor: "#b0c4de"}
+  *tweet_id {label: "bigint, not null"}
+  tweet_text {label: "varchar, not null, check(length(tweet_text) between 1 and 140)"}
```
こちら`text`のようが良いですね。

yamaryo416 commented on twitter_erd/twitter.er:
```diff
@@ -0,0 +1,71 @@
+[tweets] {bgcolor: "#b0c4de"}
+  *tweet_id {label: "bigint, not null"}
+  tweet_text {label: "varchar, not null, check(length(tweet_text) between 1 and 140)"}
+  tweet_user_id {label: "bigint, not null, FK(users)"}
```
こちら`user_id`の方が良いですね。

yamaryo416 commented on twitter_erd/twitter.er:
```diff
@@ -0,0 +1,71 @@
+[retweets] {bgcolor: "#dda0dd"}
+  *retweet_id {label: "bigint, not null"}
+  tweet_id {label: "bigint, not null, FK(tweets)"}
+  retweet_user_id {label: "bigint, not null, FK(users)"}
```
こちら `user_id`

yamaryo416 commented on twitter_erd/twitter.er:
```diff
@@ -0,0 +1,71 @@
+[bookmarks] {bgcolor: "#ffe4e1"}
+  *bookmark_id {label: "bigint, not null"}
+  tweet_id {label: "bigint, not null, FK(tweets)"}
+  bookmark_user_id {label: "bigint, not null, FK(users)"}
```
こちら`user_id`

yamaryo416 commented on twitter_erd/twitter.er:
```diff
@@ -0,0 +1,71 @@
+[favorites] {bgcolor: "#ffefd5"}
+  *favorite_id {label: "bigint, not null"}
+  tweet_id {label: "bigint, not null, FK(tweets)"}
+  favorite_user_id {label: "bigint, not null, FK(users)"}
```
こちら`user_id`

john-2132 committed:
レビューで指摘いただいた内容の反映
・主キーが１つのテーブルの主キー名はid
・外部キーが１つのテーブルの外部キー名は参照するテーブルの単数形_id
・tweet_textとdm_textをtextへ変更

john-2132 committed:
pdfの生成と、外部キー名の修正漏れがあったので再度プッシュ

john-2132 commented:
ご指摘ありがとうございます！
内容反映、修正いたしました。

yamaryo416 reviewed (approved):
LGTM!
